Nodejs/Product/TestAdapter/TestFrameworks/Jasmine/jasmine.js

```diff
@@ -107,38 +107,44 @@ function enumerateSpecs(suite, testList, testFile) {
  * @param {string} discoverResultFile
  * @param {string} projectFolder
  */
-function find_tests(testFileList, discoverResultFile, projectFolder) {
-    return new Promise(resolve => {
-        var Jasmine = detectJasmine(projectFolder);
-        if (!Jasmine) {
-            return resolve();
-        }
-        var jasmineInstance = initializeJasmine(Jasmine, projectFolder);
-        setSpecFilter(jasmineInstance, _ => false);
-
-        var testList = [];
-        testFileList.split(";").forEach((testFile) => {
-            try {
-                jasmineInstance.specDir = "";
-                jasmineInstance.specFiles = [];
-                jasmineInstance.addSpecFiles([testFile]);
-                jasmineInstance.loadSpecs();
-
-                var topSuite = jasmineInstance.env.topSuite();
-                enumerateSpecs(topSuite, testList, testFile);
-            }
-            catch (ex) {
-                //we would like continue discover other files, so swallow, log and continue;
-                console.error("Test discovery error:", ex, "in", testFile);
+async function find_tests(testFileList, discoverResultFile, projectFolder) {
+    var Jasmine = detectJasmine(projectFolder);
+    if (!Jasmine) {
+        return;
+    }
+    
+    var jasmineInstance = initializeJasmine(Jasmine, projectFolder);
+    setSpecFilter(jasmineInstance, _ => false);
+
+    var testList = [];
+    for (var testFile of testFileList.split(";")) {
+        try {
+            jasmineInstance.specDir = "";
+            jasmineInstance.specFiles = [];
+            jasmineInstance.addMatchingSpecFiles([testFile]);
```

Comment:
Will this create a compatibility issue with the previous version or this method also exists pre version 4?

Reply:
Updated to check if it exists and fall back to the old function if it doesn't.